File: src/main/java/net/matees/arcade/mobrush/settings/MaxMobCount.java
package net.matees.arcade.mobrush.settings;

import org.bukkit.Material;

import net.matees.settings.IntegerSetting;

public class MaxMobCount extends IntegerSetting {
    private Integer maxMobCount = 16;
    private static final MaxMobCount INSTANCE = new MaxMobCount();

    private MaxMobCount() {
    }

    public static MaxMobCount getInstance() {
        return INSTANCE;
    }

    @Override
    public String getName() {
        return "Max Mob Count";
    }

    @Override
    public String getDescription() {
        return "Max amount of mobs which spawn (if random is enabled, this is the range)";
    }

    @Override
    public Integer getSetting() {
        return maxMobCount;
    }

    @Override
    public void setSetting(Integer setting) {
        this.maxMobCount = setting;
    }

    @Override
    public Material getMenuItemMaterial() {
        return Material.ZOMBIE_HEAD;
    }

    @Override
    public int getMenuItemSlot() {
        return 12;
    }

}

File: src/main/java/net/matees/arcade/mobrush/settings/MaxMobCount.kt
package net.matees.arcade.mobrush.settings

import net.matees.settings.IntegerSetting
import org.bukkit.Material

class MaxMobCount private constructor() : IntegerSetting() {
    private var maxMobCount = 1

    override val name: String
        get() = "Max Mob Count"

    override val description: String
        get() = "Max amount of mobs which spawn (if random is enabled, this is the max range)"

    override var setting: Int?
        get() = maxMobCount
        set(setting) {
            if (setting != null) {
                this.maxMobCount = setting
            }
        }

    override val menuItemMaterial: Material
        get() = Material.ZOMBIE_HEAD

    override val menuItemSlot: Int
        get() = 12

    companion object {
        val instance: MaxMobCount = MaxMobCount()
    }
}
